### ISSUE TITLE
Use mongomock instead of MongoBox for testing 

### ISSUE THREAD

**Issue by danielballan** [bluesky/databroker#621]
Look at https://github.com/mongomock/mongomock.

**Comment by danielballan** [comment]
I am seeing errors out of MongoBox locally. Example in details box below. I see that @gwbischof saw the same issue and reported it upstream in July along with a PR to fix it, https://github.com/theorm/mongobox/pull/17. There was an initial response but no follow-up since. The last MongoBox release was in October 2018.

The mongomock library has an order of magnitude more forks and looks more active.

<details>

```
_____________________________________________________________________________ test_export_size_smoke[intake_mongo] ______________________________________________________________________________

broker_factory = <function broker_factory.<locals>.<dictcomp>.<lambda> at 0x7f0f243b0b80>, RE = <bluesky.run_engine.RunEngine object at 0x7f0f8ce40340>
tmpdir = local('/tmp/pytest-of-dallan/pytest-11/test_export_size_smoke_intake_1')

    def test_export_size_smoke(broker_factory, RE, tmpdir):
        from ophyd import sim
>       db1 = broker_factory()

databroker/tests/test_broker.py:728: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
databroker/tests/conftest.py:69: in <lambda>
    return {k: lambda: v(request) for k, v in param_map.items()}[request.param]
databroker/tests/utils.py:41: in build_intake_mongo_backed_broker
    box.start()
../../../miniconda3/envs/py38/lib/python3.8/site-packages/mongobox/mongobox.py:84: in start
    self._wait_till_started()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <mongobox.mongobox.MongoBox object at 0x7f0f8cf7f3a0>

    def _wait_till_started(self):
        attempts = 0
        while True:
            if self.process.poll() is not None:  # the process has terminated
>               with open(self.log_path) as log_file:
E               FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpcpdcy2r9/mongodb.log'

../../../miniconda3/envs/py38/lib/python3.8/site-packages/mongobox/mongobox.py:90: FileNotFoundError

```

</details>

**Comment by gwbischof** [comment]
The mongobox owner will merge our stuff, we probably need to email him to get his attention. Mongomock seems interesting

**Comment by danielballan** [comment]
I like that mongomock is all in memory.

**Comment by danielballan** [comment]
Closed by #619. I feel open to moving back to MongoBox if https://github.com/theorm/mongobox/pull/17 is merged and released someday, but on the other hand mongomock may run faster and be more convenient (no need to run mongo to run the tests) as long as it mocks mongo sufficiently accurately. We could also have fixtures for each.
